### ISSUE TITLE
Multiple proxies problem

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#1354]
## Description

```
      Reverse tunnel    +----------+
                        |          |
+----------------------->  Proxy 1 |
                        |          |
                        +----------+



      Client connection +----------+
                        |          |
 +---------------------->  Proxy 2 |
                        |          |
                        +----------+
```

Whenever reverse tunnel  connects to one proxy and the client connect to another, we have a problem when the cluster appears offline.

## Solution

The proposed solution is to forward client connection from Proxy 2 to Proxy 1 in this case:

```
      Reverse tunnel    +----------+
                        |          |
+----------------------->  Proxy 1 |
                        |          |
                        +----+-----+
                             ^
                             |  Forward connection to Proxy 1
                             |
      Client connection +----+-----+
                        |          |
 +---------------------->  Proxy 2 |
                        |          |
                        +----------+
```

As a part of this work we are going to improve bookeeping of reverse tunnels by adding state (what proxy is currently holding the connection) to the reverse tunnel state.

**Comment by iuriaranda** [comment]
Is this fixed in the new 2.4 release?

**Comment by klizhentas** [comment]
@iuriaranda yes
